02-peer-protocol.md

```diff
@@ -590,6 +593,15 @@ For a timed out or route-failed HTLC:
    * [2:len]
    * [len:reason]
 
+For a unparsable HTLC:
+
+1. type: 133 (`update_fail_malformed_htlc`)
+2. data:
+   * [8:channel-id]
+   * [8:id]
+   * [4:failure-code]
+   * [32:sha256-of-onion]
```

Comment:
The `onion` refers to the full serialized onionpacket in the `update_add_htlc`, not any specific part of the onion right? Alternatively we could use the HMAC internal to the onionpacket, but we'd need to hand it one layer up.